### ISSUE TITLE
Dart flags in quotes causes error

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#114622]
If I run a command like
```flutter run --dart-flags="--max_profile_depth 1"```

It returns an error like 
```[VERBOSE-3:switches.cc(441)] Encountered disallowed Dart VM flag: "--max_profile_depth 1"```
and then just hangs.

`max_profile_depth` is an allowed flag, as seen here https://github.com/flutter/engine/blob/main/shell/common/switches.cc#L57.

I believe the issue happens on iOS physical devices and macOS. I don't have a physical device to test with yet, but I was able to recreate issue on macOS. iOS Simulator does not have this issue. 

I believe issue is flags here are wrapped in quotes, so it gets double wrapped.
https://github.com/flutter/flutter/blob/d40f6d1e681f17e3b7b7290afb4c9b4175b976e7/packages/flutter_tools/lib/src/ios/devices.dart#L357
It becomes something like
```
--args --enable-dart-profiling --disable-service-auth-codes --dart-flags=""--max_profile_depth 1"" --enable-checked-mode --verify-entry-points
```

The simulator equivalent does not wrap in quotes https://github.com/flutter/flutter/blob/d40f6d1e681f17e3b7b7290afb4c9b4175b976e7/packages/flutter_tools/lib/src/ios/simulators.dart#L444
And if I change the simulator equivalent to be match the devices one, it will also fail with the disallowed error.

**Comment by darshankawar** [comment]
Thanks for the report @vashworth 
I ran the command on physical iOS device and on macOS desktop and on both platforms, it compiled and launched the app without any issues.

<details>
<summary> iOS console log </summary>

```

dhs@Dhss-MacBook-Pro test_app % flutter run --dart-flags="--max_profile_depth 1"
Launching lib/main.dart on Darshan's iphone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
Running pod install...                                           1,552ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        12.6s
Xcode build done.                                           31.7s
Installing and launching...                                        37.2s
Syncing files to device Darshan's iphone...                         81ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on Darshan's iphone is available at: http://127.0.0.1:51489/E2Hd3Bqwbu4=/
[VERBOSE-2:FlutterObservatoryPublisher.mm(138)] Could not register as server for FlutterObservatoryPublisher, permission denied. Check your 'Local Network' permissions for this app in the Privacy section of the system Settings.
[VERBOSE-2:FlutterObservatoryPublisher.mm(138)] Could not register as server for FlutterObservatoryPublisher, permission denied. Check your 'Local Network' permissions for this app in the Privacy section of the system Settings.
fopen failed for data file: errno = 2 (No such file or directory)
Errors found! Invalidating cache...
The Flutter DevTools debugger and profiler on Darshan's iphone is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:51489/E2Hd3Bqwbu4=/
fopen failed for data file: errno = 2 (No such file or directory)
Errors found! Invalidating cache...

```

</details>

<details>
<summary> macOS console log </summary>

```

dhs@Dhss-MacBook-Pro newtest % flutter run --dart-flags="--max_profile_depth 1"
Multiple devices found:
macOS (desktop) • macos  • darwin-x64     • macOS 12.2.1 21D62 darwin-x64
Chrome (web)    • chrome • web-javascript • Google Chrome 107.0.5304.87
[1]: macOS (macos)
[2]: Chrome (chrome)
Please choose one (To quit, press "q/Q"): 1
Running "flutter pub get" in newtest...                             5.7s
Launching lib/main.dart on macOS in debug mode...
Running pod install...                                             25.1s
In file included from /Users/dhs/Documents/NCFlutter/newtest/macos/Pods/FirebaseFirestore/Firestore/core/src/local/local_store.cc:44:
/Users/dhs/Documents/NCFlutter/newtest/macos/Pods/FirebaseFirestore/Firestore/core/src/util/set_util.h:39:11: warning: parameter 'comparator' not found in the function declaration [-Wdocumentation]
 * @param comparator - The comparator for the elements in before and after.
          ^~~~~~~~~~
 * @param comparator - The comparator for the elements in before and after.
          ^~~~~~~~~~
          cmp
1 warning generated.
In file included from /Users/dhs/Documents/NCFlutter/newtest/macos/Pods/FirebaseFirestore/Firestore/core/src/local/leveldb_index_manager.cc:44:
/Users/dhs/Documents/NCFlutter/newtest/macos/Pods/FirebaseFirestore/Firestore/core/src/util/set_util.h:39:11: warning: parameter 'comparator' not found in the function declaration [-Wdocumentation]
 * @param comparator - The comparator for the elements in before and after.
          ^~~~~~~~~~
 * @param comparator - The comparator for the elements in before and after.
          ^~~~~~~~~~
          cmp
1 warning generated.
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_core-1.24.0/macos/Classes/FLTFirebaseCorePlugin.m:229:50: warning: incompatible pointer to integer conversion sending 'NSNumber * _Nonnull __strong' to parameter of type 'BOOL' (aka 'signed char') [-Wint-conversion]
    [firebaseApp setDataCollectionDefaultEnabled:enabled];
                                                 ^~~~~~~
In module 'FirebaseCore' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_core-1.24.0/macos/Classes/FLTFirebasePlugin.h:10:
    BOOL dataCollectionDefaultEnabled;
         ^
1 warning generated.
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:101:72: warning: sending 'FLTFirebaseAuthPlugin *__strong' to parameter of incompatible type 'id<FLTFirebasePlugin> _Nonnull'
    [[FLTFirebasePluginRegistry sharedInstance] registerFirebasePlugin:self];
                                                                       ^~~~
In module 'firebase_core' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:7:
- (void)registerFirebasePlugin:(id<FLTFirebasePlugin> _Nonnull)firebasePlugin;
                                                               ^
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:78:17: warning: class 'FLTFirebaseAuthPlugin' does not conform to protocols 'MultiFactorUserHostApi' and 'MultiFactoResolverHostApi' [-Wprotocol]
@implementation FLTFirebaseAuthPlugin {
                ^
@implementation FLTFirebaseAuthPlugin {
                ^
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:904:3: warning: 'ASAuthorizationAppleIDProvider' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
  ASAuthorizationAppleIDProvider *appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
@interface ASAuthorizationAppleIDProvider : NSObject <ASAuthorizationProvider>
           ^
  ASAuthorizationAppleIDProvider *appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:904:55: warning: 'ASAuthorizationAppleIDProvider' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
  ASAuthorizationAppleIDProvider *appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
@interface ASAuthorizationAppleIDProvider : NSObject <ASAuthorizationProvider>
           ^
  ASAuthorizationAppleIDProvider *appleIDProvider = [[ASAuthorizationAppleIDProvider alloc] init];
                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:906:3: warning: 'ASAuthorizationAppleIDRequest' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
  ASAuthorizationAppleIDRequest *request = [appleIDProvider createRequest];
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
@interface ASAuthorizationAppleIDRequest : ASAuthorizationOpenIDRequest
           ^
  ASAuthorizationAppleIDRequest *request = [appleIDProvider createRequest];
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:909:32: warning: 'ASAuthorizationScopeFullName' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
    [requestedScopes addObject:ASAuthorizationScopeFullName];
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
AS_EXTERN ASAuthorizationScope const ASAuthorizationScopeFullName API_AVAILABLE(ios(13.0), macos(10.15), tvos(13.0), watchos(6.0));
                                     ^
    [requestedScopes addObject:ASAuthorizationScopeFullName];
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:912:32: warning: 'ASAuthorizationScopeEmail' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
    [requestedScopes addObject:ASAuthorizationScopeEmail];
                               ^~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
AS_EXTERN ASAuthorizationScope const ASAuthorizationScopeEmail API_AVAILABLE(ios(13.0), macos(10.15), tvos(13.0), watchos(6.0));
                                     ^
    [requestedScopes addObject:ASAuthorizationScopeEmail];
                               ^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:917:3: warning: 'ASAuthorizationController' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
  ASAuthorizationController *authorizationController =
  ^~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
@interface ASAuthorizationController : NSObject
           ^
  ASAuthorizationController *authorizationController =
  ^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:918:9: warning: 'ASAuthorizationController' is only available on macOS 10.15 or newer [-Wunguarded-availability-new]
      [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[ request ]];
        ^~~~~~~~~~~~~~~~~~~~~~~~~
In module 'AuthenticationServices' imported from /Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/Public/FLTFirebaseAuthPlugin.h:14:
@interface ASAuthorizationController : NSObject
           ^
      [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[ request ]];
        ^~~~~~~~~~~~~~~~~~~~~~~~~
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/firebase_auth-3.11.2/macos/Classes/FLTFirebaseAuthPlugin.m:924:13: warning: unused function 'handleAppleAuthResult' [-Wunused-function]
static void handleAppleAuthResult(FLTFirebaseAuthPlugin *object, id arguments, FIRAuth *auth,
            ^
10 warnings generated.
/Users/dhs/.pub-cache/hosted/pub.dartlang.org/cloud_firestore-3.5.1/macos/Classes/FLTFirebaseFirestorePlugin.m:455:29: warning: comparison of distinct pointer types ('FIRQuery *' and 'NSNull * _Nonnull') [-Wcompare-distinct-pointer-types]
  if (query == nil || query == [NSNull null]) {
                      ~~~~~ ^  ~~~~~~~~~~~~~
1 warning generated.
warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/App' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/Info.plist' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/AssetManifest.json' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/FontManifest.json' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/NOTICES.Z' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/fonts/MaterialIcons-Regular.otf' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/isolate_snapshot_data' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/kernel_blob.bin' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/shaders/ink_sparkle.frag' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/App.framework/Versions/A/Resources/flutter_assets/vm_snapshot_data' is located outside of the allowed root paths.

warning: Stale file '/Users/dhs/Documents/NCFlutter/newtest/build/macos/Build/Products/Debug/FlutterMacOS.framework/Versions/A/FlutterMacOS' is located outside of the allowed root paths.



/Users/dhs/Documents/NCFlutter/newtest/macos/Pods/Pods.xcodeproj: warning: The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.7, but the range of supported deployment target versions is 10.9 to 12.2. (in target 'leveldb-library' from project 'Pods')
Building macOS application...                                           
Syncing files to device macOS...                                   237ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:50895/SJeCJAgyLR0=/
The Flutter DevTools debugger and profiler on macOS is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:50895/SJeCJAgyLR0=/

```

</details>

I am on latest stable version of the framework. 

<details>
<summary> stable flutter doctor -v </summary>

```
[✓] Flutter (Channel stable, 3.3.7, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.3.7 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e99c9c7cd9 (2 days ago), 2022-11-01 16:59:00 -0700
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by vashworth** [comment]
Thank you for looking into this! At the time of making this issue, I had assumed that the simulator and physical iOS launch arguments would be handled the same, but I've since discovered that's not the case. I've described it more [here](https://github.com/flutter/flutter/pull/114628#issuecomment-1304180836). 

As for macOS, I believe I was getting the error because of the way I was running it in VSCode. I tried it again just in a terminal and I don't get the error.

So after more looking into it, I don't think it's a bug so I'll close this.
